src/NumberNode.js

```diff
@@ -4,22 +4,29 @@
   class NumberNode extends TextureGen.BaseNode {
     constructor(id) {
       super(id, 'Number', []);
-      this.value = {r: 255, g: 255, b: 255, a: 255};
 
       var input = document.createElement('input');
       input.type = 'number';
       this.input = input;
       var that = this;
-      input.addEventListener('change', function() {
+      input.addEventListener('input', function() {
         that.setValue(input.value);
       });
       this.domNode.appendChild(input);
+
+      this.value = this.input.value = 1;
     }
 
     setValue(value) {
       this.value = value;
+
+      var start = this.input.selectionStart,
+          end = this.input.selectionEnd;
+
       this.input.value = value;
 
+      this.input.setSelectionRange(start, end);
```

Comment:
I see, for cursor location (as you mentioned). I didn't think it did anything at first, since I only tested by clicking these:

![image](https://cloud.githubusercontent.com/assets/578029/14621208/85c4ca6e-05c1-11e6-9cc2-6ca7b5461b46.png)